kcqrs-client/src/main/kotlin/com/clouway/kcqrs/client/HttpEventStore.kt

```diff
@@ -0,0 +1,130 @@
+import com.google.api.client.util.Key
+import java.io.IOException
+import java.net.URL
+
+/**
+ * HttpEventStore is an implementation of EventStore which uses REST api for storing and retrieving of events.
+ *
+ * @author Miroslav Genov (miroslav.genov@clouway.com)
+ */
+class HttpEventStore(private val endpoint: URL,
+                     private val requestFactory: HttpRequestFactory) : EventStore {
+
+    override fun saveEvents(aggregateType: String, events: List<EventPayload>, saveOptions: SaveOptions): SaveEventsResponse {
+        val aggregateId = saveOptions.aggregateId
+
+        val requestEvents = events.map { EventPayloadDto(it.kind, it.timestamp, it.identityId, it.data.payload.toString(Charsets.UTF_8)) }
+
+        try {
+            val request = requestFactory.buildPostRequest(
+                    GenericUrl(endpoint.toString() + "/v1/aggregates"),
+                    JsonHttpContent(GsonFactory.getDefaultInstance(), SaveEventsRequestDto(aggregateId, saveOptions.version, requestEvents))
+            )
+            request.throwExceptionOnExecuteError = false
+
+            val response = request.execute()
+
+            if (response.isSuccessStatusCode) {
+                val resp = response.parseAs(SaveEventsResponseDto::class.java)
+                return SaveEventsResponse.Success(aggregateId, resp.version)
+            }
+
+            if (response.statusCode == HttpStatusCodes.STATUS_CODE_CONFLICT) {
+                val resp = response.parseAs(SaveEventsResponseDto::class.java)
+                return SaveEventsResponse.EventCollision(aggregateId, resp.version)
+            }
+
+        } catch (ex: IOException) {
+            return SaveEventsResponse.ErrorInCommunication
+        }
+
+        return SaveEventsResponse.Error("Generic Error")
+    }
+
+    override fun getEvents(aggregateId: String): GetEventsResponse {
+        val request = requestFactory.buildGetRequest(GenericUrl(endpoint.toString() + "/v1/aggregates/$aggregateId"))
+        request.throwExceptionOnExecuteError = false
+        try {
+            val response = request.execute()
+
+            // Aggregate was not found and no events cannot be returned
+            if (response.statusCode == HttpStatusCodes.STATUS_CODE_NOT_FOUND) {
+                return GetEventsResponse.AggregateNotFound
+            }
+
+            if (response.isSuccessStatusCode) {
+                val resp = response.parseAs(GetEventsResponseDto::class.java)
+
+                val events = resp.events.map { EventPayload(it.kind, it.timestamp, it.identityId, Binary(it.payload)) }
+
+                return GetEventsResponse.Success(aggregateId, resp.aggregateType, null, resp.version, events)
+            }
+
+            return GetEventsResponse.Error("got unknown error")
+
+        } catch (ex: IOException) {
+            return GetEventsResponse.ErrorInCommunication
+        }
+    }
+
+    override fun revertLastEvents(aggregateId: String, count: Int): RevertEventsResponse {
```

Comment:
This means that the whole aggregate root need to be returned, which doesn't sound like a good idea. I used PATH instead of DELETE as it touches the content of the aggregate.